x-pack/plugins/actions/server/builtin_action_types/lib/get_custom_agents.test.ts

```diff
@@ -31,15 +36,41 @@ describe('getCustomAgents', () => {
     configurationUtilities.getProxySettings.mockReturnValue({
       proxyUrl: ':nope: not a valid URL',
       proxyRejectUnauthorizedCertificates: false,
+      proxyBypassHosts: undefined,
+      proxyOnlyHosts: undefined,
     });
-    const { httpAgent, httpsAgent } = getCustomAgents(configurationUtilities, logger);
+    const { httpAgent, httpsAgent } = getCustomAgents(configurationUtilities, logger, targetUrl);
     expect(httpAgent).toBe(undefined);
     expect(httpsAgent instanceof HttpsAgent).toBeTruthy();
   });
 
   test('return default agents for undefined proxy options', () => {
-    const { httpAgent, httpsAgent } = getCustomAgents(configurationUtilities, logger);
+    const { httpAgent, httpsAgent } = getCustomAgents(configurationUtilities, logger, targetUrl);
     expect(httpAgent).toBe(undefined);
     expect(httpsAgent instanceof HttpsAgent).toBeTruthy();
   });
+
+  test('returns non-proxy agents for matching proxyBypassHosts', () => {
+    configurationUtilities.getProxySettings.mockReturnValue({
+      proxyUrl: 'https://someproxyhost',
+      proxyRejectUnauthorizedCertificates: false,
+      proxyBypassHosts: new Set([targetHost]),
+      proxyOnlyHosts: undefined,
+    });
+    const { httpAgent, httpsAgent } = getCustomAgents(configurationUtilities, logger, targetUrl);
+    expect(httpAgent instanceof HttpProxyAgent).toBeFalsy();
+    expect(httpsAgent instanceof HttpsProxyAgent).toBeFalsy();
+  });
```

Comment:
Should we add tests for the reverse? When hostname is not in `proxyBypassHosts` and when hostname is not in `proxyOnlyHosts`?

Reply:
I've added tests for these cases in commit https://github.com/elastic/kibana/pull/95365/commits/7626ecef1e66fea2db637e1fa38939b328217acd